Feature/cabinet client

QpYujin opened the pull request.

QpYujin committed:
Switch over to basic client

QpYujin committed:
Client code including code from kapture-client appropriated for cabinet for next stage

QpYujin committed:
Remove unneeded code

QpYujin committed:
Update CabinetClient.java

Removed testing key

wfowlks commented on src/main/java/com/kaleido/cabinet/client/CabinetClientConfiguration.java:
```diff
@@ -0,0 +1,92 @@
+/*
```
We don't need the copyright header since there is a License.md file stating the fair use of this project at the root of the project

wfowlks commented on src/main/java/com/kaleido/cabinet/client/CabinetClientConfiguration.java:
```diff
@@ -0,0 +1,92 @@
+
+    private CabinetClientProperties cabinetClientProperties;
+
+
```
please remove the extra space

wfowlks commented on src/main/java/com/kaleido/cabinet/client/CabinetClientConfiguration.java:
```diff
@@ -0,0 +1,92 @@
+        return retryTemplate;
+    }
+
+ 
```
Please remove the space

wfowlks commented on src/main/java/com/kaleido/cabinet/client/CabinetClientConfiguration.java:
```diff
@@ -0,0 +1,92 @@
+                restTemplate, retryTemplate, PlateMap.class);
+    }
+
+ 
```
Please remove the excess space

wfowlks commented on src/main/java/com/kaleido/cabinet/client/CabinetClientHTTPException.java:
```diff
@@ -0,0 +1,43 @@
+/*
```
no need for the header

wfowlks commented on src/main/java/com/kaleido/cabinet/client/CabinetClientProperties.java:
```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2019. Kaleido Biosciences. All Rights Reserved
```
No need for the header

wfowlks commented on src/main/java/com/kaleido/cabinet/client/CabinetClientConfiguration.java:
```diff
@@ -0,0 +1,92 @@
+   
+
+    /**
+     * Configuration for Kapture Client Retry template.  When a request returns an exception related to 502 or 503
```
```suggestion
     * Configuration for Cabinet Client Retry template.  When a request returns an exception related to 502 or 503
```

wfowlks commented on src/main/java/com/kaleido/cabinet/client/CabinetClientProperties.java:
```diff
@@ -0,0 +1,115 @@
+public class CabinetClientProperties {
+
+    /*
+     * The properties username, password and base assume a localhost server where those credentials would be expected
```
I think this configuration information would be better served by being in the README.md file at the root of this project under a section called configuration or something similar, so that we can have in a central place all environment variables that need to be set or overridden.

wfowlks commented on src/main/java/com/kaleido/cabinet/client/CabinetResponseErrorHandler.java:
```diff
@@ -0,0 +1,48 @@
+        } else if (httpResponse.getStatusCode().series() == CLIENT_ERROR){
+            throw new HttpClientErrorException(httpResponse.getStatusCode(), httpResponse.getStatusText());
+        }
+
```
Please remove the excess space

wfowlks commented on src/main/java/com/kaleido/cabinet/client/ParameterizedTypeImpl.java:
```diff
@@ -0,0 +1,39 @@
+/*
```
We don't need the copyright, (additionally the year is incorrect)

wfowlks reviewed (changes requested):
This is a good first pass, some changes mostly semantic & documentation based

QpYujin committed:
Spacing formatting and removing copyright header

QpYujin committed:
Added url for dev environment

QpYujin committed:
merged with barcodes

QpYujin committed:
Merging with latest changes

QpYujin committed:
Merge branch 'feature/cabinetClient' of github.com:Kaleido-Biosciences/fetch into feature/cabinetClient

QpYujin committed:
Shifted configuration to readme

QpYujin committed:
Moved setting platemap status for complete api

When calling api to set platemap to complete status, the actual setting is done in cabinet

QpYujin committed:
Undo the most recent commit

Set status in fetch, and cabinet will update based on it

wfowlks commented on README.md:
```diff
@@ -34,6 +34,15 @@ docker build -t "${CI_REGISTRY_IMAGE}:latest" .
+values of username, password and the base URL you can develop against a local host test server, remote test server
+using the same code base that you would then deploy to production and connect to a production server. Each time
+you only need to over-ride the relevant variables using a environment statements such as
+'export CABINET_CLIENT_USERNAME=my_user_name' 'export CABINET_CLIENT_PASSWORD=my_password'
```
Suggested formatting: 
```
export CABINET_CLIENT_USERNAME=my_user_name
export CABINET_CLIENT_PASSWORD=my_password
export CABINET_CLIENT_BASE=https://myserver.com/api/
```

QpYujin replied:
got it

wfowlks reviewed (changes requested):
Please update the readme to make the exports easier to copy/paste, and then I think this looks good for a first pass

QpYujin committed:
Updated readme file to make it easier to read

wfowlks reviewed (approved):
good work!
